tesseract/tesseract_collision/include/tesseract_collision/core/common.h

```diff
@@ -119,24 +119,21 @@ inline ContactResult* processResult(ContactTestData& cdata,
   if (cdata.req.is_valid && !cdata.req.is_valid(contact))
     return nullptr;
 
+  if (!(contact.distance < cdata.collision_margin_data.getPairCollisionMarginData(key.first, key.second)))
+    return nullptr;
+
   if (!found)
   {
     ContactResultVector data;
     if (cdata.req.type == ContactTestType::FIRST)
     {
-      if (contact.distance < cdata.collision_margin_data.getPairCollisionMarginData(key.first, key.second))
-      {
-        data.emplace_back(contact);
-        cdata.done = true;
-      }
+      data.emplace_back(contact);
+      cdata.done = true;
     }
     else
     {
-      if (contact.distance < cdata.collision_margin_data.getPairCollisionMarginData(key.first, key.second))
-      {
-        data.reserve(100);  // TODO: Need better way to initialize this
-        data.emplace_back(contact);
-      }
+      data.reserve(100);  // TODO: Need better way to initialize this
+      data.emplace_back(contact);
     }
 
     return &(cdata.res->insert(std::make_pair(key, data)).first->second.back());
```

Comment:
The bug was due to this line which would access the data object back element where for the condition that `contact.distance` is not less than the margin` the vector would be empty causing a segfault. Moving the check to the top and return null if not valid prevented this from occurring.